main/src/ca/uwaterloo/flix/language/errors/TypeError.scala

```diff
@@ -292,4 +292,29 @@ object TypeError {
       vt << Underline("Tip:") << " Add an instance for the type." << NewLine
     }
   }
+
+  /**
+    * An error indicating that the main function's scheme is incorrect.
+    *
+    * @param actualScheme   the erroneous function's scheme.
+    * @param expectedScheme the scheme the main function is expected to have.
+    * @param loc            the location where the error occurred.
+    */
+  case class IllegalMain(actualScheme: Scheme, expectedScheme: Scheme, loc: SourceLocation) extends TypeError {
```

Comment:
In this case, I think we usually refer to `actualScheme` as `declaredScheme`, right?